Create ui-c8y-1021-56-0-generate-widget-schemas-for-validation-purposes-graft-release-cd-8646.md (#3138)
SoftwareAG/c8y-docs: content/change-logs/application-enablement/ui-c8y-1021-56-0-generate-widget-schemas-for-validation-purposes-graft-release-cd-8646.md

---
date: ""

> title: Generate schemas from types and interfaces in build process

Reviewer: ```suggestion
title: Generating schemas from TypeScript types and interfaces in application build process
```

> title: Generate widget schemas for validation purposes. [GRAFT][release/cd]  (#8646)

Reviewer 2: ```suggestion
title: Generate schemas from types and interfaces in build process
```

product_area: Application enablement & solutions
change_type:
  - value: change-QHu1GdukP
    label: Feature
component:
  - value: component-YbYJ3gLU_
    label: Web SDK
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: MTM-62204
version: 1021.56.0
---

> Generate widget schemas for validation purposes. [GRAFT][release/cd]  (#8646)

Reviewer 2: This feature allows to generate schema from types and interfaces from project and then import them later in runtime. It is possible thanks to custom webpack plugin that collects all the types imported with prefix `c8y-schema-loader` during app build process. It opens possibility to e.g. validating widget configuration in runtime (Typescript types are not present in built app).

> Generate widget schemas for validation purposes. [GRAFT][release/cd]  (#8646)

Reviewer 2: ```suggestion
This feature allows to generate schema from types and interfaces from project and then import them later in runtime. It is possible thanks to custom webpack plugin that collects all the types imported with prefix `c8y-schema-loader` during app build process. It opens possibility to e.g. validating widget configuration in runtime (Typescript types are not present in built app).
```

> This feature allows to generate schema from types and interfaces from project and then import them later in runtime. It is possible thanks to custom webpack plugin that collects all the types imported with prefix `c8y-schema-loader` during app build process. It opens possibility to e.g. validating widget configuration in runtime (Typescript types are not present in built app).

Reviewer 3: ```suggestion
This feature enables the generation of schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the built application).This improvements will support upcoming advanced dashboarding features for improving copying and pasting functionality.
```

> It is now possible to generate schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the build application).This improvements support upcoming advanced dashboard features for improved copying and pasting functionality.

Reviewer: ```suggestion
It is now possible to generate schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the built application).This improvements support upcoming advanced dashboard features for improved copying and pasting functionality.
```

> This feature enables the generation of schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the built application).This improvements will support upcoming advanced dashboarding features for improving copying and pasting functionality.

Reviewer: ```suggestion
It is now possible to generate schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the build application).This improvements support upcoming advanced dashboard features for improved copying and pasting functionality.
```

> It is now possible to generate schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the built application).This improvements support upcoming advanced dashboard features for improved copying and pasting functionality.

Reviewer 3: ```suggestion
It is now possible to generate schemas from TypeScript types and interfaces in your project, which can then be imported during runtime. A custom webpack plugin makes this functionality possible by collecting all types imported with the `c8y-schema-loader` prefix during the application build process. The enhancement enables runtime validation capabilities, such as validating widget configurations (since TypeScript types are not retained in the built application). This supports upcoming advanced features such as dashboard export and import.
```
